src/reason_pprint_ast.ml

```diff
@@ -2804,7 +2804,7 @@ class printer  ()= object(self:'self)
       | Pcstr_record r -> [self#record_declaration r]
       | Pcstr_tuple [] -> []
       | Pcstr_tuple l when polymorphic -> List.mapi ampersand_helper l
-      | Pcstr_tuple l -> [makeTup (List.map self#non_arrowed_non_simple_core_type l)]
+      | Pcstr_tuple l -> [makeTup (List.map self#core_type l)]
```

Comment:
Here's why this works. With the new syntax, all the `args`, are already inside of a safely guarded place like `Constructor(here, andHere)`. Compare that to the previous syntax `Constructor here andHere`. In the previous syntax, we needed to require that we print "non-arrowed" types for `here`, and `andHere` to avoid something like `Constructor a=>b c=>d`. In the new syntax, we don't care if `here` and `andHere` have unguarded arrow types like `a=>b` because they're safely separated by commas.